### PR TITLE
Catch if cfg->protocol not set

### PR DESCRIPTION
**Pull request by needlerp**

### CODE REVIEW

File: src/esp32/esp32_wifi.c [hunk 1]
```diff
@@ -326,13 +326,15 @@ bool mgos_wifi_dev_sta_setup(const struct mgos_config_wifi_sta *cfg) {
   } else {
     tcpip_adapter_dhcpc_start(TCPIP_ADAPTER_IF_STA);
   }
-  const char *protocol = cfg->protocol;
-  r = esp32_wifi_protocol_setup(WIFI_IF_STA, protocol);
-  if (r != ESP_OK) {
-    LOG(LL_ERROR, ("Failed to set STA protocol: %s", esp_err_to_name(r)));
-    goto out;
+  if (!mgos_conf_str_empty(cfg->protocol)) {
+    const char *protocol = cfg->protocol;
```
Comment (rojer) [inline src/esp32/esp32_wifi.c]: looks like this is used only once and is not needed

File: src/esp32/esp32_wifi.c [hunk 2]
```diff
@@ -326,13 +326,15 @@ bool mgos_wifi_dev_sta_setup(const struct mgos_config_wifi_sta *cfg) {
...
+    r = esp32_wifi_protocol_setup(WIFI_IF_STA, protocol);
+    if (r != ESP_OK) {
+      LOG(LL_ERROR, ("Failed to set STA protocol: %s", esp_err_to_name(r)));
+      goto out;
+    }
   }
-
+
```
Comment (rojer) [inline src/esp32/esp32_wifi.c]: remove trailing space

File: src/esp32/esp32_wifi.c (hunk 1 above)
Comment (needlerp) [reply src/esp32/esp32_wifi.c]: removed const char *protocol

File: src/esp32/esp32_wifi.c (hunk 2 above)
Comment (needlerp) [reply src/esp32/esp32_wifi.c]: Removed

**Pull request merged**